streaming/src/test/java/org/apache/spark/streaming/JavaAPISuite.java

```diff
@@ -1818,26 +1815,25 @@ public Integer call(String s) throws Exception {
   // InputStream functionality is deferred to the existing Scala tests.
   @Test
   public void testSocketTextStream() {
-      JavaReceiverInputDStream<String> test = ssc.socketTextStream("localhost", 12345);
+    ssc.socketTextStream("localhost", 12345);
   }
 
-  @Test
-  public void testSocketString() {
-
-    class Converter implements Function<InputStream, Iterable<String>> {
```

Comment:
Declaring classes inside methods is super weird in Java, to me, even if it's allowed. It seemed clearer to establish it as a simple static inner class. I guess I'm not even clear it needs to be non-anonymous, and so on reflection, maybe it just can be an anonymous inner class, which is not weird.